plugins/inputs/sqlserver/sqlqueriesV2.go

```diff
@@ -1222,16 +1222,28 @@ ELSE
 `
 
 const sqlServerRequestsV2 string = `
-SET NOCOUNT ON; 
+SET NOCOUNT ON;
+DECLARE 
+	 @SqlStatement AS nvarchar(max)
+	,@MajorMinorVersion AS int = CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') as nvarchar),4) AS int) * 100 + CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') as nvarchar),3) AS int)
+
+-- 2008R2 and before doesn't have open_transaction_count in sys.dm_exec_sessions
+DECLARE @Columns as nvarchar(max) = ''
+IF @MajorMinorVersion >= 1200  
+	SET @Columns = ',s.open_transaction_count as open_transaction '
+ELSE
+	SET @Columns = ',r.open_transaction_count as open_transaction'
+
+SET @SqlStatement = N'
 SELECT  blocking_session_id into #blockingSessions FROM sys.dm_exec_requests WHERE blocking_session_id != 0
 create index ix_blockingSessions_1 on #blockingSessions (blocking_session_id)
-SELECT	
-'sqlserver_requests' AS [measurement]
-, REPLACE(@@SERVERNAME,'\',':') AS [sql_instance]
+SELECT 
+''sqlserver_requests'' AS [measurement]
+, REPLACE(@@SERVERNAME,''\'','':'') AS [sql_instance]
 , DB_NAME() as [database_name]
 , s.session_id
 , ISNULL(r.request_id,0) as request_id
-, DB_NAME(r.database_id) as session_db_name
+, DB_NAME(s.database_id) as session_db_name
```

Comment:
s.database_id is not available in SQL 2008 and R2, it should be r.database_id

But since this is dynamic SQL now you can just pass it in the column parameter
```TSQL
IF @MajorMinorVersion >= 1200  
	SET @Columns = ',s.open_transaction_count as open_transaction
	,DB_NAME(s.database_id) as session_db_name'
ELSE
	SET @Columns = ',r.open_transaction_count as open_transaction
	,DB_NAME(r.database_id) as session_db_name'
```

after this the PR is approved to me

Reply:
Dang it sorry  moving target should install a 2008/R2 instance at some point. should be done now.